tests/sentry/relay/config/test_metric_extraction.py

```diff
@@ -1274,20 +1598,22 @@ def test_get_metric_extraction_config_with_no_spec(default_project: Project) ->
         assert config["metrics"][0].get("condition") is None
 
 
-# XXX: This may have bugs, therefore, use with caution
-def _on_demand_spec_from_widget(widget: DashboardWidgetQuery) -> OnDemandMetricSpec:
+def _on_demand_spec_from_widget(
+    project: Project, widget: DashboardWidgetQuery
+) -> OnDemandMetricSpec:
     field = widget.aggregates[0] if widget.aggregates else ""
-    return OnDemandMetricSpec(
+    return fetch_on_demand_metric_spec(
```

Comment:
I can now use this shared function to reduce differences between the code and the tests.